src/board/ui.ts

```diff
@@ -624,3 +628,81 @@ export class PinUI {
 
   dispose() {}
 }
+
+interface RadioConfig {
+  maxPayload: number;
+  queue: number;
+  group: number;
+}
+
+export class RadioUI {
+  private rxQueue: (Uint8Array | undefined)[] | undefined;
+  private config: RadioConfig | undefined;
+
+  peek(): Uint8Array | undefined {
+    return this.rxQueue![0];
+  }
+
+  pop() {
+    this.rxQueue!.shift();
+  }
+
+  send(data: Uint8Array) {
+    window.parent.postMessage(
+      {
+        kind: "radio_output",
+        data,
+      },
+      "*"
+    );
+  }
+
+  receive(data: Uint8Array) {
+    if (this.rxQueue?.length === this.config?.queue) {
+      // Drop the message as the queue is full.
+    } else {
+      // Add extra information to make a radio packet in the expected format
+      // rather than just data. Clients must prepend \x01\x00\x01 if desired.
+      const len = data.length;
+      const size =
+        1 + // len
+        len +
+        1 + // RSSI
+        4; // time
+      const rssi = 127; // What's a reasonable value?
+      const time = 0; // Function needs moving so we can call it.
+
+      const packet = new Uint8Array(size);
+      packet[0] = len;
+      packet.set(data, 1);
+      packet[1 + len] = rssi;
+      packet[1 + len + 1] = time & 0xff;
+      packet[1 + len + 2] = (time >> 8) & 0xff;
+      packet[1 + len + 3] = (time >> 16) & 0xff;
+      packet[1 + len + 4] = (time >> 24) & 0xff;
+
+      this.rxQueue!.push(packet);
+    }
+  }
+
+  updateConfig(config: RadioConfig) {
+    // Maybe needs to be conditional?
+    this.disable();
+    this.enable(config);
+  }
+
+  enable(config: RadioConfig) {
+    // Can you call more than once?
```

Comment:
Needs checking.

Reply:
You can call it more than once and it disabled/enables with the new config, reallocating the receive buffer.
That matches our behaviour.